Sources/MongoDriver/MongoDriver.swift

```diff
@@ -109,29 +117,36 @@ extension MongoKitten.Database : Fluent.Driver, Connection {
                     throw Error.unsupported
                 }
             case .group(let relation, let filters):
-                if relation == .and {
-                    subQuery = try makeQuery(filters, method: .and)
-                } else {
-                    subQuery = try makeQuery(filters, method: .or)
+                switch relation {
+                case .and:
+                    subQuery = try self.makeQuery(filters, method: .and)
+                case .or:
+                    subQuery = try self.makeQuery(filters, method: .or)
                 }
-            case .subset(let key, let scope, let values):
+            case .subset(var key, let scope, let values):
+
+                key = eval(key)
+
                 switch scope {
                 case .in:
                     subQuery = MKQuery(aqt: AQT.in(key: key, in: values))
                 case .notIn:
                     subQuery = MKQuery(aqt: AQT.not(AQT.in(key: key, in: values)))
                 }
             }
-            
+
             if query.makeDocument().count == 0  {
                 query = subQuery
-            } else if method == .and {
-                query = query && subQuery
             } else {
-                query = query || subQuery
+                switch method {
+                case .and:
+                    query = query && subQuery
```

Comment:
MongoKitten now has the `&=` operator, I think it's a good opportunity to use that here now.